### PR TITLE
Registration endpoint

### PR DESCRIPTION
**Pull request by alekott**

### CODE REVIEW

File: src/main/java/com/example/chat/groupchatbackend/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    public ResponseEntity register(@RequestBody User user){
+        Optional<User> optionalUser = userRepository.findById(user.getId());
+        if (optionalUser.isPresent()) {
+            return new ResponseEntity(HttpStatus.MULTI_STATUS);
+        } else {
+            userRepository.save(user);
+            return new ResponseEntity(HttpStatus.OK);
+        }
+    }
+}
```
Comment (mateusz-siwiec) [inline src/main/java/com/example/chat/groupchatbackend/controllers/UserController.java]: GREAT JOB !

**Review by mateusz-siwiec — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -35,6 +35,20 @@
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<!-- https://mvnrepository.com/artifact/org.springframework.data/spring-data-jpa -->
```
Comment (michal-marciniec) [inline pom.xml]: redundant line

File: src/main/java/com/example/chat/groupchatbackend/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import com.example.chat.groupchatbackend.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.util.Optional;
+
+public class UserController {
```
Comment (michal-marciniec) [inline src/main/java/com/example/chat/groupchatbackend/controllers/UserController.java]: please test with postman

File: src/main/java/com/example/chat/groupchatbackend/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @PostMapping("/registration")
+    public ResponseEntity register(@RequestBody User user){
+        Optional<User> optionalUser = userRepository.findById(user.getId());
+        if (optionalUser.isPresent()) {
+            return new ResponseEntity(HttpStatus.MULTI_STATUS);
```
Comment (michal-marciniec) [inline src/main/java/com/example/chat/groupchatbackend/controllers/UserController.java]: `BAD_REQUEST` with message

**Review by michal-marciniec — APPROVED** [review]

**Pull request merged**